### ISSUE TITLE
Auth requested url redirect problem

### ISSUE THREAD

**Issue by monsurhoq** [ADmad/CakePHP-HybridAuth#107]
I am facing redirect problem for google. For facebook, it's working. But for google it's display `redirect_uri_mismatch` error.

My AppController setup as below:

>         $this->loadComponent ('TinyAuth.Auth', [
>             'autoClearCache' => true,
>             'authenticate' => [
>                  'Form' => [
>                     'finder' => 'auth'
>                  ],
>                  'ADmad/HybridAuth.HybridAuth' => [
>                     'fields' => [
>                         'provider' => 'provider',
>                         'openid_identifier' => 'openid_identifier',
>                         'email' => 'email'
>                     ],
>                     'profileModel' => 'ADmad/HybridAuth.SocialProfiles',
>                     'profileModelFkField' => 'user_id',
>                     'userModel' => 'Users',
>                     'hauth_return_to' => null,
>                     'finder' => 'auth'
>                 ]
>             ],
>             'loginAction' => [
>                 'controller' => 'Users',
>                 'action' => 'login',
>                 'prefix' => false,
>                 'plugin' => false
>             ],
>             'loginRedirect' => [
>                 'controller' => 'Users',
>                 'action' => 'index',
>                 'prefix' => false,
>                 'plugin' => false
>             ],
>             'logoutRedirect' => [
>                 'controller' => 'Users',
>                 'action' => 'login',
>                 'prefix' => false,
>                 'plugin' => false
>             ],
>             'flash' => [
>                 'element' => 'error',
>                 'key' => 'auth'
>             ],
>             'authorize' => []
>          ]);

My request URL as below:

>          $this->Form->postLink (
>              __('Login with Facebook'),
>               ['controller' => 'Users', 'action' => 'login', 'provider' => 'Facebook', 'redirect' => $this->request->query('redirect')]
>          );
>          $this->Form->postLink (
>               __('Login with Google'),
>              ['controller' => 'Users', 'action' => 'login', 'provider' => 'Google', 'redirect' => $this->request->query('redirect')]
>          );

I am using CakePHP version 3.4.10 and  dmad/cakephp-hybridauth version 4.1.1

**Comment by ADmad** [comment]
Yeah Google gets cranky if the callback URL isn't exactly the same as the one configured for it's Oauth2 service access. We would need to remove the appending of querystring to callback URL and instead rely on session to keep track of the redirect URL.
